sc-memory/sc-memory/sc_link.hpp

```diff
@@ -100,6 +100,11 @@ class ScLink
                          "You use incorrect type. Use IsType<>() to check it");
     }
     ScStreamPtr const stream = m_ctx.GetLinkContent(m_addr);
+
+    // Check for empty content.
+    if (stream && !stream->IsValid())
+      return Type();
```

Comment:
My proposal is to do such logic:
```cpp
ScStreamPtr ScMemoryContext::GetLinkContent(ScAddr const & addr)
{
  SC_ASSERT(IsValid(), ());

  sc_stream * s = nullptr;
  if (sc_memory_get_link_content(m_context, *addr, &s) != SC_RESULT_OK || s == nullptr)
    return ScStreamPtr();

  return std::make_unique<ScStream>(s);
}
```

Let say, that main function can return `nullptr`. Then we will check that across whole code. The main idea there is to say, that we don't generate `ScStream` for empty contents. Let return pointer will be null.
Because when it is not null developer will have crashes for any function call for returned `ScStream` because internal pointer will be empty. 
Let developer do usage like this:
```cpp
ScStreamPtr const stream = m_ctx.GetLinkContent(m_addr);
if (stream)
{
}
```

instead of 
```cpp
ScStreamPtr const stream = m_ctx.GetLinkContent(m_addr);

if (stream->IsValid())
{
}
```

Understanding, that pointer can be null is native for any developer.

Also this requires change in current function. We should not throw an exception:
```cpp

    if (!stream)
      return Type();

    Type result;
    if (!Stream2Value(stream, result))
    {
        SC_THROW_EXCEPTION(utils::ExceptionCritical,
                           "Error to get value of " + std::to_string(m_addr.Hash()));
    }

    return result;
```

Reply:
By doing this we won't be able to differentiate between GetLinkContent() returning a null pointer because of empty content and it returning a null pointer because of some error.

Reply:
Error throws an exception

Reply:
I mean an error inside `sc_memory_get_link_content()`, which causes it not to return `SC_RESULT_OK`

Reply:
Yeah. We can check that and throw an exception